Letterbook.Docs/_blog/2024-11-tk/moderation-tooling-research.md

```diff
@@ -0,0 +1,130 @@
+
+Finally, most moderators are not experts in their softare. Moderation in the fediverse is largely a volunteer effort, and training tends to be on the job. Communicating with a moderator on another instance about how to get at certain information poses challenges due to differences in how server software's interfaces work. If you're trying to help another moderator out, you can't assume they're using the same software as you, or even the same *version* of softare as you. Their interface might be different, and this poses a communication challenge.
+
+# Obstacles
+
+There are a handful of obstacles to improving moderation tooling, but the biggest one is getting people with influence to care. Large technology projects require collaboration and vision. In private companies, this typically rests in the hands of someone with a "Product" title, who acts as the arbiter of values that go into making the software what it is. Open source projects tend to instead have figureheads who spearheaded a project's technical foundation and steer the project's direction and values as it gains in popularity.
+
+In talking with people doing moderation work, I got an overwhelming sense of frustration that people with influence over the software and protocols haven't prioritized this work. If we are to provide a safe federated space equivalent to popular commercial platforms, we need the people involved in both the protocol and server software to pay attention.
+
+Improving the protocol is the more important challenge -- the fediverse is not *just* Mastodon, and ultimately any initiative to improve trust and safety between server instances will require improvements and clarifications to the protocol so the plethora of server softwares people use can continue to interoperate. This sort of collaboration and consensus-building takes time and a variety of perspectives.
+
+I was also surprised to find that data retention policy poses a set of obstacles to moderation work in the fediverse. Federating posts over ActivityPub is, to some degree, a data processing agreement. Retention is a very delicate and thorny issue with a lot of legal, social, and ethical consequences; one that many volunteer or amateur administrators are not ready to deal with. The GDPR is the elephant in the room, but it can also provide some guidance on good defaults for server software to set for its retention policies.
+
+# Recommendations
+
+Based on these conversations, I can make a handful of recommendations to developers who want to help improve trust & safety in the fediverse:
+
+1. Get involved in improving the protocol. I get it -- consensus-building is hard work, and talking with people isn't as fun as writing code. It is important work, it is important to push back against people with influence who don't value this work, and I believe it will yield the best long-term results. 
```

Comment:
There's actually a lot of people doing this. The problem is that the channels for doing it aren't empowered to make the kinds of changes that are necessary. And also, people with no stake in any real implementation are free to come in and derail any progress that ever gets made, which happens all the time.

I don't know if that changes anything about this bullet point. I guess I just wanted to be sure you have that context.

Reply:
I think they care. They're just busy and frustrated at never being able to make progress. I mean, I know I am. What I wish would happen is implementers would get together and work on specific interoperability issues somewhat in private. At least not in a way that has any obligation to consider proposals that aren't supported by real world implementations. Even if that excludes me, for now.

Reply:
I was talking about documentation with Erin Kisane earlier, and the idea of reference samples for the kinds of messages software will send and expect to receive came up. I think that would help a lot. It's very actionable, and it would give maintainers on other software concrete things to discuss. Taking these incredibly abstract conversations about the right way to structure AP messages into the realm of concrete reality would be a great thing to do.